File: sdk/examples/client/getting_started.rs
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

//! This examples shows how to get the node info.
//!
//! ```sh
//! cargo run --release --example client_getting_started
//! ```

use iota_sdk::client::Client;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::builder()
        .with_node("https://api.testnet.shimmer.network")? // Insert your node URL here
        .finish()
        .await?;

    let node_info = client.get_node_info().await?;
    println!("Node Info: {node_info:?}");

    Ok(())
}

File: bindings/nodejs/examples/client/getting-started.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, ClientError, initLogger } from '@iota/sdk';

// Run with command:
// yarn run-example ./client/getting-started.ts

// In this example we will get information about the node
async function run() {
    initLogger();
    const client = await Client.create({
        nodes: ['https://api.testnet.shimmer.network'],
    });

    try {
        const info = (await client.getNodeInfo()).info;
        console.log(info);
    } catch (error) {
        if (
            error instanceof ClientError &&
            error.name === 'healthyNodePoolEmpty'
        ) {
            console.error(
                'No healthy node available, please provide a healthy one.',
            );
        } else {
            console.error(error);
        }
    }
}

void run().then(() => process.exit());
